### PR TITLE
[Legend] - Icons use Symbols through SymbolGenerator API

### LINKED ISSUE #967: Legends should support arbitrary symbol shapes
Currently legends only draw circles as the legend icon. Maybe it's good to have an API point to set a function used to draw the icon?
```
iconShape(iconDrawer:(width: number, height: number, container: D3.Selection) => any)
```

### PR DESCRIPTION
**Pull request by bluong**
The icons next to the domain are no longer circles but instead symbols.

These can be set through a symbolGenerator() getter/setter API which defaults to the D3 circle symbol

Fixes #967

### CODE REVIEW

File: src/components/legend.ts [hunk 1]
```diff
@@ -269,11 +275,12 @@ export module Component {
...
       });
 
-      entries.select("circle")
-          .attr("cx", layout.textHeight / 2)
-          .attr("cy", layout.textHeight / 2)
-          .attr("r",  layout.textHeight * 0.3)
-          .attr("fill", (value: string) => this._scale.scale(value) );
+      entries.select("path").attr("d", this.symbolGenerator())
+                            .attr("transform", "translate(" + (layout.textHeight / 2) + "," + layout.textHeight / 2 + ") " +
+                                               "scale(" + (layout.textHeight * 0.3 / 50) + ")")
```
Comment (jtlan) [inline src/components/legend.ts]: Can we try to cut down on the magic numbers? I know the 0.3 was from before, but we should save off the 50 somewhere, probably in `SymbolGenerators`

File: src/components/legend.ts [hunk 2]
```diff
@@ -11,6 +11,10 @@ export module Component {
      * The css class applied to each legend entry
      */
     public static LEGEND_ENTRY_CLASS = "legend-entry";
+    /**
+     * The css class applied to each legend symbol
+     */
+    public static LEGEND_SYMBOL_CLASS = "legend-symbol";
```
Comment (jtlan) [inline src/components/legend.ts]: We should note that people should update their CSS selectors for this from `<circle>` to `.legend-symbol`.

File: src/components/legend.ts [hunk 3]
```diff
@@ -294,6 +301,18 @@ export module Component {
                       self._writer.write(value, maxTextLength, self.height(), writeOptions);
                     });
     }
+
+    public symbolGenerator(): SymbolGenerator;
+    public symbolGenerator(symbolGenerator: SymbolGenerator): Legend;
+    public symbolGenerator(symbolGenerator?: SymbolGenerator): any {
```
Comment (jtlan) [inline src/components/legend.ts]: Docs?

File: test/components/legendTests.ts [hunk 4]
```diff
@@ -156,11 +156,11 @@ describe("Legend", () => {
...
       var text = (<any> legend)._content.select("text");
-      var circle = (<any> legend)._content.select("circle");
+      var symbol = (<any> legend)._content.select("." + Plottable.Component.Legend.LEGEND_SYMBOL_CLASS);
       var textHeight = Plottable._Util.DOM.getBBox(text).height;
-      var circleHeight = Plottable._Util.DOM.getBBox(circle).height;
-      assert.operator(circleHeight, "<", textHeight, "icons too small: iconHeight < circleHeight");
-      assert.operator(circleHeight, ">", textHeight / 2, "icons too big: iconHeight / 2 > circleHeight");
+      var symbolHeight = symbol.node().getBoundingClientRect().height;
+      assert.operator(symbolHeight, "<", textHeight, "icons too small: iconHeight < circleHeight");
+      assert.operator(symbolHeight, ">", textHeight / 2, "icons too big: iconHeight / 2 > circleHeight");
```
Comment (jtlan) [inline test/components/legendTests.ts]: Change the string and the name of this method to generalize?

File: src/components/legend.ts (hunk 2 above)
Comment (bluong) [inline src/components/legend.ts]: For less fragility, should we name these things icons?

Comment (jtlan) [inline src/components/legend.ts]: "icon" and "symbol" are about the same.

**Comment by crmorford** [comment]
So excited to release this!

**Comment by bluong** [comment]
screen shot 2015-03-13 at 2 19 34 pm
 Release pic

**Pull request merged**